lib/session.js

```diff
@@ -110,7 +115,9 @@ var doSignin = function(lookup, req, res, onSuccess, onFail) {
 			if (user && (!passwordCheck || scmp(passwordCheck, hash(user.password)))) {
 				postHookedSigninWithUser(user, req, res, onSuccess, onFail);
 			} else {
-				onFail(err || new Error('Incorrect user or password'));
+				res.clearCookie('keystone.uid');
```

Comment:
It falls with error 500 if calls `onFail`. I didn't found function of update cookies when it expires.
Error occurs if the user has cookies from the old session. I think the better way is to delete it.
In this `if` statement `else` works only if `user` is defined. And if it defined `err` is null.